### ISSUE TITLE
stabilizerstate has no field of "phases"

### ISSUE THREAD

**Issue by hongyehu** [QuantumSavory/QuantumClifford.jl#161]
Hi Stefan, 
  I am reading the doc of QuantumClifford.jl and I cannot run the following example:
  ```
s = S"-XXX
             +ZZI
             -IZZ";
newstate, anticomindex, result = project!(s, P"ZII")
if isnothing(result)
    newstate.phases[anticomindex] = rand([0x0,0x2])
end
```
I have printed the fields of stabilizerstates, and I only see `tab` and no `phases`. Is this a version issue?
Many thanks!

**Comment by Krastanov** [comment]
Thanks for letting me know! This is due to slightly outdated documentation (this seems to be the only code snippet on that page that is not automatically verified when the code runs).

The "proper" way to get a view on the state is `phases(newstate)[anticomindex] = rand([0x0,0x2])`.

I added the following explanation to the documentation: Of course, this is a rather cumbersome way to run a simulation, so we also provide
`projectrand!` which does the necessary randomization automatically,
for cases where you do not need the fine grained control of `project!`.
There is also `projectZ!`, `projectZrand!` and corresponding X and Y methods
if you want to measure single qubit Paulis.

Fixed here: https://github.com/QuantumSavory/QuantumClifford.jl/pull/163/files#diff-8d089f016719a5a91ad06586127054c9fd5df94eb6dc881af7fc0c749980aa2aL315
